### PR TITLE
Added nexus5x to wrapper.

### PR DESCRIPTION
**Pull request by ziquanyu**
added files for nexus5x.

### CODE REVIEW

**Comment by ziquanyu** [comment]
@emilwestergren Can you review this when you have time? Thanks!

**Comment by emilwestergren** [comment]
Can you rebase this and adapt the code to the changes? Check `Android.mk` for Nexus 6. Make sure to test it on the device.

**Comment by ziquanyu** [comment]
yes, will fix it after lunch.

File: HAL/HAL3/CameraHALWrapper.cpp [hunk 1]
```diff
@@ -195,14 +195,14 @@ int CameraHALWrapper::openCameraDevice(struct hw_device_t **hw_device, camera_mo
 camera3_device_ops_t CameraHALWrapper::mCameraOps = {
 	initialize: CameraHALWrapper::initialize,
 	configure_streams: CameraHALWrapper::configure_streams,
-#if CAMERA_DEVICE_API_VERSION == CAMERA_DEVICE_API_VERSION_3_0
+#if CAMERA_DEVICE_API_VERSION >= CAMERA_DEVICE_API_VERSION_3_0
```
Comment (emilwestergren) [inline HAL/HAL3/CameraHALWrapper.cpp]: This is not correct, it should only be valid for HAL 3.0 I think.

File: HAL/HAL3/CameraHALWrapper.cpp [hunk 2]
```diff
@@ -195,14 +195,14 @@ int CameraHALWrapper::openCameraDevice(struct hw_device_t **hw_device, camera_mo
...
 	register_stream_buffers: CameraHALWrapper::register_stream_buffers,
 #else
 	register_stream_buffers: NULL,
 #endif
 	construct_default_request_settings: CameraHALWrapper::construct_default_request_settings,
 	process_capture_request: CameraHALWrapper::process_capture_request,
-#if CAMERA_DEVICE_API_VERSION == CAMERA_DEVICE_API_VERSION_3_0
+#if CAMERA_DEVICE_API_VERSION >= CAMERA_DEVICE_API_VERSION_3_0
```
Comment (emilwestergren) [inline HAL/HAL3/CameraHALWrapper.cpp]: Same here.

Comment (ziquanyu) [inline HAL/HAL3/CameraHALWrapper.cpp]: you are right, it's fixed. Tested with latest changes on 5x and nexus6. They both work well.

File: nexus5x/Android.mk [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    libcamera_client \
+    libgui \
+    libhardware \
+    libcamera_metadata \
+	libandroid \
+    libvidhance_android32
+
+#Android version
+ANDROID_VERSION_MAJOR=6
+ANDROID_VERSION_MINOR=1
```
Comment (emilwestergren) [inline nexus5x/Android.mk]: Should be 0 right?

File: nexus5x/Android.mk [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+HAL_FOLDER = HAL3
+ifeq ($(DEVICE_VERSION_MAJOR), 1)
+	HAL_FOLDER = HAL1
+endif
+
+LOCAL_SRC_FILES := \
+		../HAL/CameraHAL.cpp \
+		../HAL/CameraWrapper.cpp \
+		../HAL/CameraWrapperFactory.cpp \
+        ../HAL/$(HAL_FOLDER)/CameraHALWrapper.cpp \
```
Comment (emilwestergren) [inline nexus5x/Android.mk]: Indentation is different from the lines above.

File: nexus5x/VidhanceProcessor.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+}
+namespace imint {
+CameraWrapper* VidhanceProcessor::createVidhanceProcessor(const char* cameraId) {
+	return new VidhanceProcessor(cameraId);
+}
+struct create_method_t CameraWrapper::createMethod = { VidhanceProcessor::createVidhanceProcessor };
+
+/* Predefined aligned width in bytes for RGBA8888 GraphicBuffers on Nexus 6 */
+#define ALIGNED_WIDTH_COUNT 10
+const int alignedWidth[ALIGNED_WIDTH_COUNT] = { 128, 256, 384, 512, 1024, 1536, 2560, 3072, 3584, 4608 };
```
Comment (emilwestergren) [inline nexus5x/VidhanceProcessor.cpp]: Have you tested that these are aligned widths for Nexus5x?

Comment (ziquanyu) [inline nexus5x/VidhanceProcessor.cpp]: yes, they are aligned for nexus5x.

File: nexus5x/VidhanceProcessor.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+inline size_t ALIGN(size_t x, size_t align) {
+	return (x + align - 1) & ~(align - 1);
+}
+namespace imint {
+CameraWrapper* VidhanceProcessor::createVidhanceProcessor(const char* cameraId) {
+	return new VidhanceProcessor(cameraId);
+}
+struct create_method_t CameraWrapper::createMethod = { VidhanceProcessor::createVidhanceProcessor };
+
+/* Predefined aligned width in bytes for RGBA8888 GraphicBuffers on Nexus 6 */
```
Comment (emilwestergren) [inline nexus5x/VidhanceProcessor.cpp]: Removed or change comment then.

**Pull request merged**